python/src/aiconfig/editor/server/server.py

```diff
@@ -160,62 +166,75 @@ def home():
         return Err(f"File does not exist: {resolved}")
-    return Ok(resolved)
+    return Ok(ValidatedPath(resolved))
 
 
-def _http_response_with_path(path_fn: Callable[[str], HttpPostResponse]) -> HttpPostResponse:
+def _validated_request_path(allow_create: bool = False) -> Result[ValidatedPath, str]:
     request_json = request.get_json()
     path = request_json.get("path", None)
-
-    validated_path = _get_validated_request_path(path)
-    match validated_path:
-        case Ok(path):
-            return path_fn(path)
-        case Err(e):
-            return HttpPostResponse(message=e, code=400)
+    return _get_validated_path(path, allow_create=allow_create)
 
 
 @app.route("/api/load", methods=["POST"])
 def load():
-    def _run_with_path(path: str) -> HttpPostResponse:
-        LOGGER.info(f"Loading AIConfig from {path}")
-        state = _get_server_state(app)
-        try:
-            state.aiconfig = AIConfigRuntime.load(path)  # type: ignore
-            return HttpPostResponse(message="Done")
-        except Exception as e:
-            return HttpPostResponse(message=f"<p>Failed to load AIConfig from {path}: {e}", code=400)
+    state = _get_server_state(app)
 
-    return _http_response_with_path(_run_with_path).to_flask_format()
+    path_val = _validated_request_path()
+    res_aiconfig = path_val.and_then(_safe_load_from_disk)
+    match res_aiconfig:
+        case Ok(aiconfig):
+            LOGGER.warning(f"Loaded AIConfig from {path_val}. This may have overwritten in-memory changes.")
+            state.aiconfig = aiconfig
+            return HttpPostResponse(message="Loaded", aiconfig=aiconfig).to_flask_format()
+        case Err(e):
+            return HttpPostResponse(message=f"Failed to load AIConfig: {path_val}, {e}", code=400, aiconfig=None).to_flask_format()
 
 
 @app.route("/api/save", methods=["POST"])
 def save():
-    def _run_with_path(path: str) -> HttpPostResponse:
-        LOGGER.info(f"Saving AIConfig to {path}")
-        state = _get_server_state(app)
-        try:
-            state.aiconfig.save(path)  # type: ignore
-            return HttpPostResponse(message="Done")
-        except Exception as e:
-            err: Err[str] = core_utils.ErrWithTraceback(e)
-            LOGGER.error(f"Failed to save AIConfig to {path}: {err}")
-            return HttpPostResponse(message=f"<p>Failed to save AIConfig to {path}: {err}", code=400)
+    state = _get_server_state(app)
+    if state.aiconfig is None:
+        return HttpPostResponse(message="No AIConfig in memory, nothing to save.", code=400, aiconfig=None).to_flask_format()
+    else:
+        aiconfig: AIConfigRuntime = state.aiconfig
 
-    return _http_response_with_path(_run_with_path).to_flask_format()
+        path_val = _validated_request_path(allow_create=True)
+        res_save: Result[HttpPostResponse, str] = result.do(
+            Ok(HttpPostResponse(message="Saved to disk", aiconfig=aiconfig))
```

Comment:
nit, don't need to return the config in the response for save. Really just need to know success/fail

Reply:
Ok, I will clean this up in a future PR, keeping in mind that you only need the config on a load or successful update, right?